kpatch-build/kpatch-build

```diff
@@ -1271,7 +1271,7 @@ for i in $FILES; do
 
 		# create-diff-object orig.o patched.o parent-name parent-symtab
 		#		     Module.symvers patch-mod-name output.o
-		"$TOOLSDIR"/create-diff-object $CDO_FLAGS "orig/$i" "patched/$i" "$KOBJFILE_NAME" \
+		"$TOOLSDIR"/create-diff-object "$CDO_FLAGS" "orig/$i" "patched/$i" "$KOBJFILE_NAME" \
```

Comment:
Something to watch out for,  quoting a string that contains multiple arguments will consolidate them into a single argument.  For example:
```bash
#!/bin/bash

foo()
{
    [[ -n "$1" ]] && echo "arg1: $1"
    [[ -n "$2" ]] && echo "arg2: $2"
    [[ -n "$3" ]] && echo "arg3: $3"
}

args="--one --two --three"

echo "foo \$args"
foo $args
echo "foo \"\$args\""
foo "$args"
```
Results in two different behaviors:
``` bash
$ ./test.sh 
foo $args
arg1: --one
arg2: --two
arg3: --three
foo "$args"
arg1: --one --two --three
```
So it might be more future proof if **$CDO_FLAGS** was an array.

Also, there is a real bug when **$CDO_FLAGS** is empty.  A slightly modified repro:
``` bash
#!/bin/bash

foo()
{
    echo "arg1: $1"
    echo "arg2: $2"
}

args=""

echo "foo \"\$args\" \"next-option\""
foo "$args" "next-option"

echo "foo \$args \"next-option\""
foo $args "next-option"
```
Results again in two different behaviors:
``` bash
$ ./test.sh 
foo "$args" "next-option"
arg1: 
arg2: next-option
foo $args "next-option"
arg1: next-option
arg2:
```
The second problem breaks builds.

Modifying to store and pass though an array should solve both problems.